### PR TITLE
let focusableHoc support focus methods in wrapped Components + add fo…

### PR DESCRIPTION
**Pull request by jonathanadler**
…cu method to button-next

### CODE REVIEW

File: packages/wix-ui-core/src/hocs/Focusable/test/FocusableHOCTestFixture.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+export enum DataHook {
+    FirstButton = 'first-button',
+    SecondButton = 'second-button',
+}
+
+const ButtonElement = withFocusable(ButtonNext);
+
+export class FocusableHOCTestFixture extends React.Component {
+    private readonly firstButtonRef: React.RefObject<React.Component>;
+    private readonly secondButtonRef: React.RefObject<React.Component>;
```
Comment (shlomitc) [inline packages/wix-ui-core/src/hocs/Focusable/test/FocusableHOCTestFixture.tsx]: I think this second ref is redundant

File: packages/wix-ui-core/src/components/button-next/button-next.tsx [hunk 2]
```diff
@@ -33,7 +37,14 @@ const _addAffix = (Affix, styleClass) =>
 class ButtonNextComponent extends React.Component<ButtonProps> {
   static defaultProps = { as: 'button', type: 'button' };
   static displayName = 'ButtonNext';
-  public innerComponentRef: React.RefObject<HTMLButtonElement | ButtonProps['as']>;
+  public innerComponentRef: React.RefObject<HTMLElement | React.ComponentType>;
+
+  focus() {
+    console.log('adler', 'button-next.tsx:43', this.innerComponentRef);
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/button-next/button-next.tsx]: Leftovers

File: packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.tsx [hunk 3]
```diff
@@ -103,14 +109,14 @@ export const withFocusable = Component => {
     onFocus = event => {
       const { onFocus } = this.props;
       onFocus
-        ? onFocus(event, { blur: this.blur, focus: this.focus })
-        : this.focus();
+        ? onFocus(event, { blur: this.blur, focus: this.markAsFocused })
+        : this.markAsFocused();
```
Comment (shlomitc) [inline packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.tsx]: Maybe the same notation should be applied to the blur function in order to stay consistent

File: packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  isFocused,
+} from 'wix-ui-test-utils/protractor';
+import {buttonNextTestkitFactory} from '../../testkit/protractor';
+
+describe('FocusableHOC', () => {
+  const testsStoryUrl = getStoryUrl('Tests', 'FocusableHOC');
+
+  beforeEach(() => browser.get(testsStoryUrl));
+
+  eyes.it(
```
Comment (shlomitc) [inline packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.e2e.ts]: Does the eyes functionality needed here?

File: packages/wix-ui-core/src/components/button-next/button-next.tsx (hunk 2 above)
Comment (jonathanadler) [reply packages/wix-ui-core/src/components/button-next/button-next.tsx]: 🤦‍♂

File: packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.e2e.ts (hunk 4 above)
Comment (jonathanadler) [reply packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.e2e.ts]: you're right, not really

File: packages/wix-ui-core/src/hocs/Focusable/test/FocusableHOCTestFixture.tsx (hunk 1 above)
Comment (jonathanadler) [reply packages/wix-ui-core/src/hocs/Focusable/test/FocusableHOCTestFixture.tsx]: It's for calling `secondButtonRef.current.focus()` (see line 24)

File: packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.tsx (hunk 3 above)
Comment (jonathanadler) [reply packages/wix-ui-core/src/hocs/Focusable/FocusableHOC.tsx]: yes

**Pull request merged**